dss-collection-nft/contracts/DSSCollection.cdc

```diff
@@ -0,0 +1,626 @@
+    // Contract Events
+    //
+    pub event ContractInitialized()
+
+    // NFT Collection Events
+    //
+    pub event Withdraw(id: UInt64, from: Address?)
+    pub event Deposit(id: UInt64, to: Address?)
+
+    // Events
+    //
+    pub event CollectionGroupCreated(
+        id: UInt64,
+        name: String,
+        typeName: String,
+        startTime: UFix64?,
+        endTime: UFix64?,
+        timeBound: Bool
+    )
+    pub event CollectionGroupClosed(id: UInt64)
+    pub event SlotCreated(
+        id: UInt64,
+        collectionGroupID: UInt64,
+        logicalOperator: String,
+        typeName: String,
+        slotType: String
+    )
+    pub event ItemAddedToSlot(
+        itemID: UInt64,
+        itemValue: UInt64,
+        collectionGroupID: UInt64
+    )
+    pub event DSSCollectionNFTMinted(
+        id: UInt64,
+        collectionGroupID: UInt64,
+        serialNumber: UInt64,
+        completedBy: String,
+        completionDate: UFix64,
+        level: UInt64
+    )
+    pub event DSSCollectionNFTBurned(id: UInt64)
+
+
+    // Named Paths
+    //
+    pub let CollectionStoragePath:  StoragePath
+    pub let CollectionPublicPath:   PublicPath
+    pub let AdminStoragePath:       StoragePath
+    pub let MinterPrivatePath:      PrivatePath
+
+    // Entity Counts
+    //
+    pub var totalSupply:                 UInt64
+    pub var nextCollectionGroupID:       UInt64
+    pub var nextSlotID:       UInt64
+
+    // Lists in contract
+    //
+    access(self) let collectionGroupByID: @{UInt64: CollectionGroup}
+    access(self) let slotByID: @{UInt64: Slot}
+
+    // A public struct to access Slot data
+    //
+    pub struct SlotData {
+        pub let id: UInt64
+        pub let collectionGroupID: UInt64
+        pub let logicalOperator: String // (AND / OR)
+        pub let typeName: String // (A.contractAddress.NFT...)
+        pub let slotType: String // (edition.id, edition.tier, play.id)
+        pub var items: {UInt64: AnyStruct}
+
+        pub fun itemIDExistsInSlot(id: UInt64): Bool {
+           return self.items.containsKey(id)
+        }
+
+        init (id: UInt64) {
+            if let slot = &DSSCollection.slotByID[id] as &DSSCollection.Slot? {
+                self.id = slot.id
+                self.collectionGroupID = slot.collectionGroupID
+                self.logicalOperator = slot.logicalOperator
+                self.typeName = slot.typeName
+                self.slotType = slot.slotType
+                self.items = slot.items
+            } else {
+                panic("slot does not exist")
+            }
+        }
+    }
+
+    // A top-level Slot with a unique ID
+    //
+    pub resource Slot {
+        pub let id: UInt64
+        pub let collectionGroupID: UInt64
+        pub let logicalOperator: String // (AND / OR)
+        pub let typeName: String // (A.contractAddress.NFT...)
+        pub let slotType: String // (edition.id, edition.tier, play.id)
+        pub var items: {UInt64: UInt64}
```

Comment:
This doesn't match items in `SlotData` above

Reply:
This got refactored into an array. Should match now.